manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,654 @@
+            getLogger().warning(e.toString());
+//            getLogger().warning(e.fi);
+        }
+        // Check if asset was found
+    }
+
+    /**
+     * Creates a new asset with the correct "hashed" Asset ID, its IMEI,
+     * in the realm the MQTT message of the device submitted,
+     * and the parsed list of attributes.
+     * //TODO: Add {@value TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME }, {@value TELTONIKA_DEVICE_RECEIVE_COMMAND_ATTRIBUTE_NAME} Attributes when creating Asset
+     *
+     * @param newDeviceId The ID of the device's Asset.
+     * @param newDeviceImei The IMEI of the device. If passed to
+     *                      {@link UniqueIdentifierGenerator#generateId(String)},
+     *                      it should always return {@code newDeviceId}.
+     * @param realm The realm to create the Asset in.
+     * @param attributes The attributes to insert in the Asset.
+     */
+    private void createNewAsset(String newDeviceId, String newDeviceImei, String realm, AttributeMap attributes, Topic topic, RemotingConnection connection) {
+
+        CarAsset testAsset = new CarAsset("Teltonika Asset "+newDeviceImei)
+                .setRealm(realm)
+                .setId(newDeviceId);
+
+        testAsset.getAttributes().add(new Attribute<>(CarAsset.IMEI, newDeviceImei));
+
+
+        //Now that the asset is created and IMEI is set, pull the packet timestamp, and then
+        //set each of the asset's attributes to have that timestamp.
+
+        attributes.get(CarAsset.LAST_CONTACT).ifPresent(dateAttr -> {
+            dateAttr.getValue().ifPresent(dateVal -> {
+                testAsset.setCreatedOn(dateVal);
+                testAsset.getAttributes().forEach(attribute -> {
+                    attribute.setTimestamp(dateVal.getTime());
+                });
+                attributes.forEach(attribute -> {
+                    attribute.setTimestamp(dateVal.getTime());
+                });
+            });
+        });
+
+
+
+        updateAsset(testAsset, attributes, topic, connection);
+    }
+
+    /**
+     * Returns list of attributes depending on the Teltonika JSON Payload.
+     * Uses the logic and results from parsing the Teltonika Parameter IDs.
+     *
+     * @param payloadContent Payload coming from Teltonika device
+     * @return Map of {@link Attribute}s to be assigned to the {@link Asset}.
+     */
+    private AttributeMap getAttributesFromPayload(String payloadContent) throws JsonProcessingException {
+        HashMap<Integer, TeltonikaParameter> params = new HashMap<>();
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            // Parse file with Parameter details
+
+            // Add each element to the HashMap, with the key being the unique parameter ID and the parameter
+            // being the value
+            params = Arrays.stream(mapper.readValue(getParameterFileString(), TeltonikaParameter[].class)).collect(Collectors.toMap(
```

Comment:
You're reading the parameter file in for every payload

Reply:
We had a conversation about how we should manage the parameter files. Would we want to read it once on `TeltonikaMQTTHandler.start()`?